### ISSUE TITLE
Seeing "already initialized constant" with respect to JSON

### ISSUE THREAD

**Issue by Fryguy** [ManageIQ/manageiq-schema#11]
After each pass of the migrations specs (i.e. after the up and after the down), I see the following warning, but I can't figure out where it's coming from.

```
/Users/jfrey/.rubies/ruby-2.4.1/lib/ruby/2.4.0/json/version.rb:4: warning: already initialized constant JSON::VERSION
/Users/jfrey/.gem/ruby/2.4.1/gems/json-2.1.0/lib/json/version.rb:4: warning: previous definition of VERSION was here
/Users/jfrey/.rubies/ruby-2.4.1/lib/ruby/2.4.0/json/version.rb:5: warning: already initialized constant JSON::VERSION_ARRAY
/Users/jfrey/.gem/ruby/2.4.1/gems/json-2.1.0/lib/json/version.rb:5: warning: previous definition of VERSION_ARRAY was here
/Users/jfrey/.rubies/ruby-2.4.1/lib/ruby/2.4.0/json/version.rb:6: warning: already initialized constant JSON::VERSION_MAJOR
/Users/jfrey/.gem/ruby/2.4.1/gems/json-2.1.0/lib/json/version.rb:6: warning: previous definition of VERSION_MAJOR was here
/Users/jfrey/.rubies/ruby-2.4.1/lib/ruby/2.4.0/json/version.rb:7: warning: already initialized constant JSON::VERSION_MINOR
/Users/jfrey/.gem/ruby/2.4.1/gems/json-2.1.0/lib/json/version.rb:7: warning: previous definition of VERSION_MINOR was here
/Users/jfrey/.rubies/ruby-2.4.1/lib/ruby/2.4.0/json/version.rb:8: warning: already initialized constant JSON::VERSION_BUILD
/Users/jfrey/.gem/ruby/2.4.1/gems/json-2.1.0/lib/json/version.rb:8: warning: previous definition of VERSION_BUILD was here
/Users/jfrey/.rubies/ruby-2.4.1/lib/ruby/2.4.0/json/common.rb:100: warning: already initialized constant JSON::NaN
/Users/jfrey/.gem/ruby/2.4.1/gems/json-2.1.0/lib/json/common.rb:100: warning: previous definition of NaN was here
/Users/jfrey/.rubies/ruby-2.4.1/lib/ruby/2.4.0/json/common.rb:102: warning: already initialized constant JSON::Infinity
/Users/jfrey/.gem/ruby/2.4.1/gems/json-2.1.0/lib/json/common.rb:102: warning: previous definition of Infinity was here
/Users/jfrey/.rubies/ruby-2.4.1/lib/ruby/2.4.0/json/common.rb:104: warning: already initialized constant JSON::MinusInfinity
/Users/jfrey/.gem/ruby/2.4.1/gems/json-2.1.0/lib/json/common.rb:104: warning: previous definition of MinusInfinity was here
/Users/jfrey/.rubies/ruby-2.4.1/lib/ruby/2.4.0/json/common.rb:129: warning: already initialized constant JSON::UnparserError
/Users/jfrey/.gem/ruby/2.4.1/gems/json-2.1.0/lib/json/common.rb:129: warning: previous definition of UnparserError was here
```

**Comment by Fryguy** [comment]
This happens on Ruby 2.3 as well

**Comment by Fryguy** [comment]
@jrafanie Thoughts?

**Comment by jrafanie** [comment]
log the caller of version.rb:4, 5, 6, 7, 8, etc...

Maybe there's a fine detail difference in how it's being required...

**Comment by jrafanie** [comment]
Such as require vs. require_relative or require with FQ path vs. relative, etc.

**Comment by Fryguy** [comment]
Figured it out.  So, I noticed this happened specifically with `rake spec:migrations:complete_up` and `rake spec:migrations:complete_down`.  Those are suspicious because they shell out to another rake task, then wait for that subprocess to complete.  However, all the do in the subprocess is a `bundle exec rake db:migrate`.  So, I tried running just that, and I don't get the warnings.  This led me to conclude that is not the rake task, but the act of shelling out.  Then, I remembered ye olde Bundle.with_clean_env, and this worked.  Bundler somehow messes with the env causing rake to use the pure stdlib json, but activerecord to use the json gem.
